ReactNativeClient/lib/joplin-database.js

```diff
@@ -873,13 +883,48 @@ class JoplinDatabase extends Database {
 		return true;
 	}
 
+	async fuzzySearchEnabled() {
+		try {
+			await this.selectOne('SELECT count(*) FROM notes_spellfix');
+		} catch (error) {
+			this.logger().warn('Fuzzy search check failed', error);
+			return false;
+		}
+		this.logger().info('Fuzzy search check succeeded');
+		return true;
+	}
+
+
 	version() {
 		return this.version_;
 	}
 
 	async initialize() {
 		this.logger().info('Checking for database schema update...');
 
+		if (shim.isElectron()) {
+			// if prod
+			try {
+				await this.loadExtension(`${process.env.LD_LIBRARY_PATH.split(':')[0]}/spellfix`);
+			} catch (error) {
+				// console.info(error);
+			}
+			// if dev
+			try {
+				await this.loadExtension('./lib/sql-extensions/spellfix');
+			} catch (error) {
+				// console.info(error);
+			}
+		} else {
+			// if dev (CLI)
+			try {
+				await this.loadExtension('./build/lib/sql-extensions/spellfix');
+			} catch (error) {
+				// console.info(error);
+			}
+		}
```

Comment:
The database should not have platform checks, instead please use dependency injection and pass to the database the relevant spellfix extension path. The mobile app won't set any path so here you should just check if a path is set - if it is, load the extension, otherwise do nothing.